### ISSUE TITLE
CCM managed control plane load balancing, wrong port for a service cloud-provider-cherry-kubernetes-external

### ISSUE THREAD

**Issue by zalmarge** [cherryservers/cloud-provider-cherry#84]
When it is opted for CCM managed control plane, a service *cloud-provider-cherry-kubernetes-external* is created:

```
Name:                     cloud-provider-cherry-kubernetes-external
Namespace:                kube-system
Labels:                   <none>
Annotations:              metallb.universe.tf/address-pool: disabled-metallb-do-not-use-any-address-pool
Selector:                 <none>
Type:                     LoadBalancer
IP Family Policy:         SingleStack
IP Families:              IPv4
IP:                       10.255.80.42
IPs:                      10.255.80.42
IP:                       5.199.174.20
LoadBalancer Ingress:     5.199.174.20
Port:                     https  443/TCP
TargetPort:               6443/TCP
NodePort:                 https  32486/TCP
Endpoints:                10.172.53.34:6443,10.172.53.43:6443,10.172.53.44:6443
Session Affinity:         None
External Traffic Policy:  Cluster
Events:                   <none>
```

From what I understand, for the most part it is constructed according to the service *kubernetes* in the cluster, but it should not set port to 443/TCP as no-one  expects for it to change from what ever was used during cluster setup (6443 by default).

At the moment, cloud provider makes the port change, which blocks users from connecting and also breaks its own health check:

```
I1217 08:30:52.610268   31316 fip_controlplane_reconciliation.go:190] handling update, endpoints: default/kubernetes
I1217 08:30:52.613721   31316 fip_controlplane_reconciliation.go:231] handling update, service: default/kubernetes
I1217 08:30:52.613901   31316 fip_controlplane_reconciliation.go:134] handling update, node: cp-1
I1217 08:30:53.292272   31316 fip_controlplane_reconciliation.go:558] doHealthCheck(): checking control plane health through ip https://5.199.174.20:6443/healthz
E1217 08:30:53.309480   31316 fip_controlplane_reconciliation.go:574] http client error during healthcheck, will try to reassign to a healthy node. err "Get "https://5.199.174.20:6443/healthz": dial tcp 5.199.174.20:6443: connect: connection refused"
I1217 08:30:53.309554   31316 fip_controlplane_reconciliation.go:577] doHealthCheck(): health check through elastic ip failed, trying to reassign to an available controlplane node
I1217 08:30:53.374186   31316 fip_controlplane_reconciliation.go:267] healthcheck node https://5.199.162.163:6443/healthz
I1217 08:30:53.618374   31316 fip_controlplane_reconciliation.go:291] control plane endpoint assigned to new server cp-1
I1217 08:30:53.618483   31316 fip_controlplane_reconciliation.go:134] handling update, node: cp-2
I1217 08:30:54.158943   31316 fip_controlplane_reconciliation.go:134] handling update, node: cp-3
I1217 08:31:52.610601   31316 fip_controlplane_reconciliation.go:190] handling update, endpoints: default/kubernetes
I1217 08:31:52.613959   31316 fip_controlplane_reconciliation.go:231] handling update, service: default/kubernetes
I1217 08:31:52.614051   31316 fip_controlplane_reconciliation.go:134] handling update, node: cp-1
I1217 08:31:53.371229   31316 fip_controlplane_reconciliation.go:558] doHealthCheck(): checking control plane health through ip https://5.199.174.20:6443/healthz
E1217 08:31:53.390792   31316 fip_controlplane_reconciliation.go:574] http client error during healthcheck, will try to reassign to a healthy node. err "Get "https://5.199.174.20:6443/healthz": dial tcp 5.199.174.20:6443: connect: connection refused"
I1217 08:31:53.390841   31316 fip_controlplane_reconciliation.go:577] doHealthCheck(): health check through elastic ip failed, trying to reassign to an available controlplane node
I1217 08:31:53.449907   31316 fip_controlplane_reconciliation.go:267] healthcheck node https://5.199.168.60:6443/healthz
I1217 08:31:56.519712   31316 fip_controlplane_reconciliation.go:291] control plane endpoint assigned to new server cp-2
```

In short, do not change the port, stick to what ever kube-apiserver is using or is set in configuration.

Oh, and by the way, as long as kube-proxy is in iptables mode, it is not a problem as there's some iptables rule (or what not, haven't thoroughly checked) that redirects anything sent to port 443 to 6443. But as soon as the mode is switched to ipvs, balancing starts working as it should with all the consequences.
